### PR TITLE
"Dropdown" component - Update the API for the "index" to use toggles as yielded components

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

After discussion with @MelSumner we decided that we wanted to try to see if a different API for the `index` element of the `Dropdown` component would make sense, in particular have also the toggles as yielded components, not just the list items.

### :hammer_and_wrench: Detailed description

In this PR I have:
- updated the “index” in Dropdown to use the toggles as yielded components
- updated the “toggles” subcomponents to expose an `onClick` callback (used in the “index”)
   - this `onClick` needs a guard to avoid errors if the parameter is not passed
      - an alternative would be to use `optional` from the [ember-composable-helpers](https://github.com/DockYard/ember-composable-helpers#optional)?
- updated the documentation page for the “Dropdown” to use the new (proposed) APIs

***

### 👀 How to review

👉 Review commit-by-commit or by files changed

Reviewer's checklist:

- [ ] +1 Percy if applicable

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202083397114692

### CODE REVIEW

**Comment by MelSumner** [comment]
cc @meirish on this one, since we discussed it yesterday. What do you think?

**Comment by didoo** [comment]
During discussion today with @MelSumner, while merging #66, we have agreed that this change in the API is OK.

File: packages/components/tests/dummy/app/templates/components/dropdown.hbs [hunk 1]
```diff
@@ -358,8 +352,8 @@
           <td>Row 2, cell 1</td>
           <td>Row 2, cell 2</td>
           <td>
-            <Hds::Dropdown @text="Overflow Options" @icon="more-horizontal" @hasChevron={{false}} as |dd|>
-
+            <Hds::Dropdown as |dd|>
+              <dd.ToggleIcon @iconName="more-horizontal" @text="Overflow Options" @hasChevron={{false}} />
```
Comment (MelSumner) [inline packages/components/tests/dummy/app/templates/components/dropdown.hbs:356]:
```suggestion
              <dd.ToggleIcon @icon="more-horizontal" @text="Overflow Options" @hasChevron={{false}} />
```

File: packages/components/tests/dummy/app/templates/components/dropdown.hbs [hunk 2]
```diff
@@ -324,12 +322,8 @@
...
       @language="markup"
       @code='
-        &lt;Hds::Dropdown
-          @icon="more-horizontal"
-          @hasChevron=&lbrace;&lbrace;false&rbrace;&rbrace;
-          @text="Overflow Options"
-          as |dd| &gt;
-
+        &lt;Hds::Dropdown as |dd|&gt;
+          &lt;dd.ToggleIcon @text="Overflow Options" @iconName="more-horizontal" @hasChevron=\{{false}} /&gt;
```
Comment (MelSumner) [inline packages/components/tests/dummy/app/templates/components/dropdown.hbs:326]:
```suggestion
          &lt;dd.ToggleIcon @text="Overflow Options" @icon="more-horizontal" @hasChevron=\{{false}} /&gt;
```

File: packages/components/tests/dummy/app/templates/components/dropdown.hbs [hunk 3]
```diff
@@ -245,8 +242,8 @@
     <nav class="dummy-nav-dropdown" aria-label="example positioned right">
       <ul class="dummy-nav-dropdown__list">
         <li class="dummy-nav-dropdown__list-item">
-          <Hds::Dropdown @toggle="toggleButton" @text="Text Toggle" as |dd|>
-
+          <Hds::Dropdown as |dd|>
```
Comment (MelSumner) [inline packages/components/tests/dummy/app/templates/components/dropdown.hbs:245]: The code sample on line 267 needs to be updated too

File: packages/components/tests/dummy/app/templates/components/dropdown.hbs [hunk 4]
```diff
@@ -476,11 +467,8 @@
     <CodeBlock
       @language="markup"
       @code='
-        &lt;Hds::Dropdown
-          @text="settings menu"
-          @icon="settings"
-          as |dd| &gt;
-
+        &lt;Hds::Dropdown as |dd|&gt;
+          &lt;dd.ToggleIcon @text="settings menu" @iconName="settings" /&gt;
```
Comment (MelSumner) [inline packages/components/tests/dummy/app/templates/components/dropdown.hbs:471]:
```suggestion
          &lt;dd.ToggleIcon @text="settings menu" @icon="settings" /&gt;
```

File: packages/components/tests/dummy/app/templates/components/dropdown.hbs [hunk 5]
```diff
@@ -495,7 +483,8 @@
     <nav class="dummy-nav-dropdown" aria-label="example row settings positioned right">
       <ul class="dummy-nav-dropdown__list">
         <li class="dummy-nav-dropdown__list-item">
-          <Hds::Dropdown @text="settings menu" @icon="settings" as |dd|>
+          <Hds::Dropdown as |dd|>
+            <dd.ToggleIcon @text="settings menu" @iconName="settings" />
```
Comment (MelSumner) [inline packages/components/tests/dummy/app/templates/components/dropdown.hbs:487]:
```suggestion
            <dd.ToggleIcon @text="settings menu" @icon="settings" />
```

File: packages/components/tests/dummy/app/templates/components/dropdown.hbs (hunk 3 above)
Comment (MelSumner) [reply packages/components/tests/dummy/app/templates/components/dropdown.hbs:245]: pushed a fix.

**Review by MelSumner — APPROVED** [review]
Tests are passing; had to tweak a few things (icon instead of iconName, a code sample or two) but OTT it was good 👍

**Pull request merged**